foreign_cc/private/cmake_script.bzl

```diff
@@ -17,9 +17,30 @@ def _escape_dquote_bash_crosstool(text):
     # We use a starlark raw string to prevent the need to escape backslashes for starlark as well.
     return text.replace('"', r'\\\\\\\"')
 
+_TARGET_OS_PARAMS = {
+    "android": {
+        "ANDROID": "YES",
+        "CMAKE_SYSTEM_NAME": "Linux",
+    },
+    "linux": {
+        "CMAKE_SYSTEM_NAME": "Linux",
+    },
+}
+
+_TARGET_ARCH_PARAMS = {
+    "aarch64": {
+        "CMAKE_SYSTEM_PROCESSOR": "aarch64",
```

Comment:
There's a little more nuance to it than this (mostly on mac) but I think this is a good enough start.